src/FluentValidation/DefaultValidatorOptions.cs

```diff
@@ -397,6 +397,20 @@ public static IRuleBuilderInitialCollection<T, TCollectionElement> Where<T, TCol
 		return rule;
 	}
 
+	/// <summary>
+	/// Applies an asynchronous filter to a collection property.
+	/// </summary>
+	/// <param name="rule">The current rule</param>
+	/// <param name="predicate">The condition</param>
+	/// <returns></returns>
+	public static IRuleBuilderInitialCollection<T, TCollectionElement> WhereAsync<T, TCollectionElement>(this IRuleBuilderInitialCollection<T, TCollectionElement> rule, Func<TCollectionElement, Task<bool>> predicate) {

```

Comment:
can use ValueTask<bool>

Reply:
The rest of the API uses `Task` and I prefer consistency